lg69t/firmware_tool.py

```diff
@@ -240,23 +273,53 @@ def main():
     args = parser.parse_args()
 
     port_name = args.port
+    p1fw_path = args.p1fw
     gnss_bin_path = args.gnss
     app_bin_path = args.app
     should_send_reboot = not args.manual_reboot
 
-    if app_bin_path is None and gnss_bin_path is None:
-        print('You must specify gnss or app files to upgrade.')
+    if p1fw_path is None and app_bin_path is None and gnss_bin_path is None:
+        print('You must specify p1fw file, gnss file, or app file to upgrade.')
         sys.exit(1)
 
     print(f"Starting upgrade on device {port_name}.")
 
-    if gnss_bin_path is not None:
+    p1fw = None
+    app_bin_fd = None
+    gnss_bin_fd = None
+    if p1fw_path is not None:
+        try:
+            p1fw = ZipFile(p1fw_path, 'r')
+        except:
+            if os.path.exists(p1fw_path):
```

Comment:
What is the case where `p1fw = ZipFile(p1fw_path, 'r')` throws an exception, but you still want to load the file? Oh I see you're supporting specifying a directory. For one thing this isn't documented in the README, for another it would probably be clearer to check `os.path.isdir(p1fw_path)` if that's your intent rather then relying on an exception.

Reply:
You can see my latest changes address this. But, if I get rid of the `try` statement, check out the default error it throws: 
```
Starting upgrade on device /dev/ttyUSB0.
Traceback (most recent call last):
  File "/home/william/firmware-tools/lg69t/firmware_tool.py", line 330, in <module>
    main()
  File "/home/william/firmware-tools/lg69t/firmware_tool.py", line 298, in main
    p1fw = ZipFile(p1fw_path, 'r')
  File "/usr/lib/python3.10/zipfile.py", line 1267, in __init__
    self._RealGetContents()
  File "/usr/lib/python3.10/zipfile.py", line 1334, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
```

So what I'm wondering is if you think I should still wrap the line `p1fw = ZipFile(p1fw_path, 'r')` in a try statement or not, or if I should just let it throw this error.

Reply:
@axlan's comment was to check first whether it is a directory. If it is, do directory stuff. Then, do zip file stuff. Using a try catch on operations that can fail with an exception is good practice, but abusing it to fork the execution path for "normal" operation is bad practice.